enhancements/project-request-limit.md

```diff
@@ -0,0 +1,392 @@
+
+## Proposal
+1. Create a `Validating` admission webhook server that provides `requestlimit.project.openshift.io/ProjectRequestLimit`.
+2. The admission webhook will be fronted by a `Service`.
+3. The webhook server is reachable through the `kube-apiserver` via the `kubernetes.default.svc service` service. 
+2. Create an operator that manages all lifecycle aspects of the admission webhook via a custom resource.
+3. Productize the operator and ship it as a `RedHat operator` via `OperatorHub`. 
+4. Provide official documentation on how to interact with the `ProjectRequestLimit` operator.
+
+### User Stories [optional]
+*Story 1*: As a cluster admin I want to enable `ProjectRequestLimit` on an OpenShift 4.x cluster so that I can limit creation of `Project`.
+
+*Story 2*: As a cluster admin I want to remove `ProjectRequestLimit` from my cluster.
+
+*Story 3*: As a cluster admin I want to be able to install `ProjectRequestLimit` in a disconnected environment.
+
+*Story 4*: As a cluster admin I want my cluster to automatically upgrade to the new version of `ProjectRequestLimit` when available.
+
+*Story 5*: As a cluster admin I want to be able to specify (at any time) limits on the number of user provisioned `Project(s)`.
+
+Notes:
+* Port `ProjectRequestLimit` plugin from 3.11 into a `Validating` admission webhook.
+* The source code is here: https://github.com/openshift/origin/blob/release-3.11/pkg/project/apiserver/admission/requestlimit/admission.go
+* Use  [generic-admission-server](https://github.com/openshift/generic-admission-server) to wire the validating admission webhook server. 
+* The operator is OLM enabled.
+* Ship `ProjectRequestLimit` as a RedHat operator via OperatorHub.
+* For productization, if we go through ART, integration work should target 4.5.
+* The operator should wire the OLM manifests accordingly to facilitate disconnected install (`relatedImages`).
+
+
+### Implementation Details/Notes/Constraints
+
+#### The Operand
+We build the webhook admission server to also be an extension API server, it becomes possible to aggregate it as a normal
+API server. An `APIService` object named `v1.projectrequestlimits.admission.project.openshift.io` makes the API group 
+`v1.projectrequestlimits.admission.project.openshift.io/v1` available within and outside of the cluster via API aggregation 
+of `kube-apiserver`. The group can be reached at `/apis/projectrequestlimits.admission.project.openshift.io/v1/validatingadmissionreviews` 
+of the `kube-apiserver`, i.e. via the `kubernetes.default.svc` service hostname inside the cluster. Below, we show how 
+we can achieve this: 
+
+* The admission webhook will be fronted by a `Service` named `admission-server`, as shown below.
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  namespace: openshift-project-request-limit-operator
+  name: admission-server
+  annotations:
+    service.alpha.openshift.io/serving-cert-secret-name: server-serving-cert
```

Comment:
don't we have beta now?

Reply:
depends on which is the minimal openshift version we want to support.

Reply:
I don't know the difference in feature between alpha and beta, we can use whatever is more appropriate.